<a href="https://colab.research.google.com/github/iMan0405/Portfolio_DT_vs_AI/blob/main/Sentiment_Analysis_for_Social_Media.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Sentiment Analysis modeli (Ijtimoiy tarmoq uchun)

### ushbu modelda twitter dataseti dan foydalanamiz va izohlarda qoldirilgan hissiyotlarni tahlil qilamiz:

Dataset haqida ma'lumot:
**textID** - matnning har bir qismi uchun noyob identifikator;
**text** - izoh matni;
**sentiment** - izohga berilgan hissiyot.

### birinchi navbatda kutubxonalarni yuklab olishimiz lozim

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

import nltk
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.model_selection import train_test_split
import string
from tqdm import tqdm
from multiprocessing import Pool
from nltk.corpus import stopwords
from nltk.stem.porter import PorterStemmer

from sklearn.svm import SVC
from sklearn.naive_bayes import BernoulliNB
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report, f1_score, roc_auc_score, roc_curve, precision_score, recall_score

import warnings
warnings.filterwarnings('ignore')

### Sentiment datasetini yuklab olamiz

In [2]:
df = pd.read_csv('/content/Tweets.csv')
df.head()

textID                                               text  \
0  cb774db0d1                I`d have responded, if I were going   
1  549e992a42      Sooo SAD I will miss you here in San Diego!!!   
2  088c60f138                          my boss is bullying me...   
3  9642c003ef                     what interview! leave me alone   
4  358bd9e861   Sons of ****, why couldn`t they put them on t...   

                         selected_text sentiment  
0  I`d have responded, if I were going   neutral  
1                             Sooo SAD  negative  
2                          bullying me  negative  
3                       leave me alone  negative  
4                        Sons of ****,  negative

### textID va selected_text ustunlarini o'chirib yuboramiz, shuningdek, yangi o'zgaruvchiga df ni saqlab olamiz

In [3]:
df.drop(['textID', 'selected_text'], axis = 1, inplace = True )
target = 'sentiment'
df.reset_index(drop = True, inplace = True)
original_df = df.copy(deep = True)
df.head()

text sentiment
0                I`d have responded, if I were going   neutral
1      Sooo SAD I will miss you here in San Diego!!!  negative
2                          my boss is bullying me...  negative
3                     what interview! leave me alone  negative
4   Sons of ****, why couldn`t they put them on t...  negative

In [5]:
print("Ma'lumot to'plamlarining o'lchamlari: ", df.shape)
df.info()

Ma'lumot to'plamlarining o'lchamlari:  (27481, 2)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27481 entries, 0 to 27480
Data columns (total 2 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   text       27480 non-null  object
 1   sentiment  27481 non-null  object
dtypes: object(2)
memory usage: 429.5+ KB


In [6]:
df.describe()

text sentiment
count                                  27480     27481
unique                                 27480         3
top      I`d have responded, if I were going   neutral
freq                                       1     11118

### yuqorida holatda df ni tahlil qildik, va text ustunida 27480 ta, sentiment ustunida 27481 ta ma'lumot bor, demak text ustunida 1 ta null qiymat borligini bildiradi, buni bartaraf qilishimiz lozim

In [7]:
df.isnull().sum()

text         1
sentiment    0
dtype: int64

In [8]:
df.dropna(inplace = True)
original_df = df.copy()

### dulikatlarni tekshirib ko'ramiz

In [9]:
df.duplicated().sum()

0